application/src/main/java/org/togetherjava/tjbot/features/basic/CoolMessagesBoardManager.java

```diff
@@ -0,0 +1,129 @@
+package org.togetherjava.tjbot.features.basic;
+
+import net.dv8tion.jda.api.EmbedBuilder;
+import net.dv8tion.jda.api.JDA;
+import net.dv8tion.jda.api.entities.Message;
+import net.dv8tion.jda.api.entities.MessageEmbed;
+import net.dv8tion.jda.api.entities.MessageReaction;
+import net.dv8tion.jda.api.entities.User;
+import net.dv8tion.jda.api.entities.channel.concrete.TextChannel;
+import net.dv8tion.jda.api.entities.emoji.Emoji;
+import net.dv8tion.jda.api.events.message.react.MessageReactionAddEvent;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import org.togetherjava.tjbot.config.Config;
+import org.togetherjava.tjbot.config.CoolMessagesBoardConfig;
+import org.togetherjava.tjbot.features.MessageReceiverAdapter;
+
+import java.awt.*;
+import java.util.Collections;
+import java.util.Optional;
+import java.util.function.Predicate;
+import java.util.regex.Pattern;
+
+/**
+ * Manager for the cool messages board. It appends highly-voted text messages to a separate channel
+ * where members of the guild can see a list of all of them.
+ */
+public final class CoolMessagesBoardManager extends MessageReceiverAdapter {
+
+    private static final Logger logger = LoggerFactory.getLogger(CoolMessagesBoardManager.class);
+    private static final Emoji REACT_EMOJI = Emoji.fromUnicode("🌟");
+    private final Predicate<String> boardChannelNamePredicate;
+    private final CoolMessagesBoardConfig config;
+
+    public CoolMessagesBoardManager(Config config) {
+        this.config = config.getCoolMessagesConfig();
+
+        boardChannelNamePredicate =
+                Pattern.compile(this.config.boardChannelPattern()).asMatchPredicate();
+    }
+
+    @Override
+    public void onMessageReactionAdd(MessageReactionAddEvent event) {
+        final MessageReaction messageReaction = event.getReaction();
+        int originalReactionsCount = messageReaction.hasCount() ? messageReaction.getCount() : 0;
+        boolean isCoolEmoji = messageReaction.getEmoji().getName().equals(REACT_EMOJI.getName());
+        long guildId = event.getGuild().getIdLong();
+        Optional<TextChannel> boardChannel = getBoardChannel(event.getJDA(), guildId);
+
+        if (boardChannel.isEmpty()) {
+            logger.warn("Tried to find the board channel in server {} but was unable to do so",
+                    guildId);
+            return;
+        }
+
+        // If the bot has already reacted to this message, then this means that
+        // the message has been quoted to the cool messages board, so skip it.
+        if (hasBotReacted(event.getJDA(), messageReaction)) {
+            return;
+        }
+
+        if (isCoolEmoji && originalReactionsCount + 1 >= config.minimumReactions()) {
+            event.retrieveMessage().queue(message -> {
+                message.addReaction(REACT_EMOJI).queue();
+
+                insertCoolMessage(boardChannel.orElseThrow(), message);
```

Comment:
boardChannel will exist at this point so will not throw because of the check on line 51?

Reply:
Nice catch! It has been changed to `boardChannel.get()` now. 🙌